_sources/Unit5-Writing-Classes/topic-5-1-parts-of-class.rst

```diff
@@ -37,49 +37,131 @@
+You can create instances of the ``Person`` class with ``new`` as in ``new
+Person()`` And you can declare variables that can hold a reference to a
+``Person`` object with ``Person variableName``.
 
-Let's create a class called Person. What would we want to know about a person?  What we want to know depends on what problem we are trying to solve.  In one situation, perhaps when creating an address book, we might want to know the person's name and phone number and email. Here's a possible Person class with instance variables, constructor, and methods.
-
-.. figure:: Figures/personClassDefn.png
-    :width: 500px
-    :align: center
-    :alt: Person
-    :figclass: align-center
+Or put it altogether to declare some variables and initialize each one with a
+reference to a new ``Person`` as shown here.
 
-    Figure 1: Person class
+.. code-block:: java
 
-Run the code below to see how it constructs 2 Person objects and fills in their data. We will explain all parts of this class in the next sections, but notice that execution always starts in the main method. When a method like the print() method is called, we run the code in the print method for that object. After a method is done, the control returns back to the next line of code in the main method. You can also see this in the |Java visualizer| (click on the link and then Forward at the bottom to run the code step by step).
+    Person ada = new Person();
+    Person charles = new Person();
+
+So what makes up the body of the class—the stuff between the ``{}``\
+s?
+
+Remember that objects have both attributes and behaviors. These correspond to
+**instance variables** and **methods** in the class definition.
+
+The first things we define in a class are usually the instance variables. They
+are called that because each instance of the class (each object) has its own set
+of variables that aren’t shared with other instances. This is what allowed
+``yertle`` and ``myrtle`` from Unit 2 to be at different positions at the same
+time; they each had their own x position and y position instance variables.
+
+The next thing we define in a class is usually its **constructors**. We’ll talk
+about writing constructors in more detail in the next section but a
+constructor’s job is to initialize the instance variables when the object is
+created. Usually that will mean they need to take arguments. (The call to ``new
+Person()`` before is to a constructor that doesn’t take any arguments. In a
+moment we’ll see that our ``Person`` constructor will actually need arguments.)
+
+The real meat of a class is in the **methods** which define the behaviors of the
+objects of that class. Recall from Unit 2 that most methods either do things
+(like the ``Turtle`` methods that moved the turtle on the screen) or return
+values like the ``getXPos`` and ``getYPos`` on ``Turtle``.
+
+When a method is called, the code runs and whenever it refers to an instance
```

Comment:
Reword? "When a method is called, the code runs and whenever it refers to an instance
variable it gets the specific values of the variable in the particular object we
called the method on" --> "When an object's method is called, the method uses the instance variable values for that object. For example, ...."